### DISCUSSION TITLE
How to disable operator= A2 auto generation?

### DISCUSSION THREAD

**Discussion by hackjackhack** [hsutter/cppfront#1042] [Q&A] [answered]
I tried to define a type with no assignment operator. Although documentation states that cpp2 doesn't need '=delete' because everything is opt-in but it seems A2 is always in effect. Is there a way to define move-ctor without move-assignment generated?

**Comment by jcanizales** [comment]
What's A2?

**Reply by gregmarr** [reply]
There's a picture that shows the versions of `operator=` that are generated based on other versions, and there are arrows labeled A1, A2, B1, B2.

**Reply by MaxSagebaum** [reply]
Which picture?

**Reply by JohelEGP** [reply]
The one at <https://github.com/hsutter/cppfront/wiki/Cpp2:-operator=,-this-&-that#operator-can-generalize-assignment-from-construction-and-move-from-copy>.
@hsutter The link to this design note seems to be gone from the wiki's home page.

**Reply by MaxSagebaum** [reply]
Thanks.

**Reply by hsutter** [reply]
> @hsutter The link to this design note seems to be gone from the wiki's home page.

Right, it moved into the documentation here: https://hsutter.github.io/cppfront/cpp2/types/#operator-can-generalize-assignment-from-construction-and-move-from-copy

Same for https://github.com/hsutter/cppfront/wiki/Cpp2-setup:-Visual-Studio

But for those who have the old link (and who see it in the navigation on the right-hand side of the wiki), I've now replaced those pages with "this page has moved" pointers to the current material. Thanks!

**Comment by JohelEGP** [comment, marked as answer]
IIRC, by manually declaring it `private`: <https://github.com/hsutter/cppfront/issues/468#issuecomment-1627565647>.
